### PR TITLE
EMS1056 vertex processing robustness

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
Fix reading past the end of the buffer causing incorrect rendering and crash later on.

### CODE REVIEW

File: lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp [hunk 1]
```diff
@@ -1010,6 +1010,11 @@ void HdVP2Mesh::Sync(
             for (size_t i = 0; i < numFaceVertexIndices; i++) {
                 const int sceneFaceVtxId = faceVertexIndices[i];
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:1012]: See line 1005, which created the buffer filled with -1, assuming the number of vertices would reflect the highest possible index. Sometimes, those indices can be greater, probably when geometry is processed and some vertices are dropped?

Whatever is the cause, we need to validate the input data when creating the rapid lookup tables.

File: lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp [hunk 2]
```diff
@@ -1010,6 +1010,11 @@ void HdVP2Mesh::Sync(
...
 
+                // Scene index is actually greater than anticipated, increase the buffer size.
+                if (sceneFaceVtxId >= _meshSharedData->_sceneToRenderingFaceVtxIds.size()) {
+                    _meshSharedData->_sceneToRenderingFaceVtxIds.resize(sceneFaceVtxId + 1, -1);
```
Comment (AramAzhari-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:1015]: Does this trigger a resize only on the first index i ?

Comment (pierrebai-adsk) [reply lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:1015]: I can trigger a resize whenever the scene face vertex ID is too large.

**Review by AramAzhari-adsk — APPROVED** [review]

**Review by AramAzhari-adsk — APPROVED** [review]

**Comment by pierrebai-adsk** [comment]
The PF passed. One MacOSX passed, all tests passed, but for some reason iyt aborted after all tests were run without reporting its results.

**Pull request merged**